unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/ECS7ComponentsPlugin.cs

```diff
@@ -1,17 +1,26 @@
 using System.Collections;
 using System.Collections.Generic;
+using System.IO;
 using DCL;
+using DCL.ECSComponents;
 using UnityEngine;
 
 public class ECS7ComponentsPlugin : IPlugin
 {
     public ECS7ComponentsPlugin()
     {
-        // When we have clear how do we want to create components in the scenes, we should register here the components
+        RegisterComponents();
+    }
+
+    public void RegisterComponents()
+    {
+        DataStore.i.ecs7.componentsFactory.AddOrReplaceComponent(ComponentID.BOX_SHAPE,
+            data => PBBoxShape.Parser.ParseFrom((byte[])data),
+            () =>  new ECSBoxShapeComponentHandler());
     }
     
     public void Dispose()
     {
-        
+        DataStore.i.ecs7.componentsFactory.RemoveComponent(ComponentID.BOX_SHAPE);
```

Comment:
The DataStore can be provided throught Dependency Injection so we can mock it and avoid singletons.

Reply:
I talked about this with Pato on Friday, I forgot that we have the non-written rule about having parameter-less constructors for plugins. So in this case this usage is correct.

Reply:
In the near future DataStore will be part of the ServiceLocator